azurerm/internal/services/advisor/data_source_advisor_recommendations.go

```diff
@@ -0,0 +1,212 @@
+package advisor
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/advisor/mgmt/2020-01-01/advisor"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	uuid "github.com/satori/go.uuid"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func dataSourceArmAdvisorRecommendations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmAdvisorRecommendationsRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"categories_filter": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(advisor.HighAvailability),
+						string(advisor.Security),
+						string(advisor.Performance),
+						string(advisor.Cost),
+						string(advisor.OperationalExcellence)}, true),
+				},
+			},
+
+			"resource_group_names_filter": azure.SchemaResourceGroupNameSetOptional(),
```

Comment:
and
```suggestion
			"filter_by_resource_groups": azure.SchemaResourceGroupNameSetOptional(),
```

Reply:
I'm also not sure its worth creating a schema function for this given its not duplicated anywhere else/

Reply:
Here I add a schema function because during the reviewing of this PR, the schema function in `resource_group.go` continues to be added. I need to change all the `ValidateFunc: validateResourceGroupName` to `ValidateFunc: ValidateResourceGroupName`

Reply:
Do you think I shall remove the schema function?